File: cooker.ts
import readline from "readline";

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let powerOne: boolean = false;
let keepWarm: boolean = false;
const maxWeight: number = 500;
let weight: number = 0;
let start: number = 0;

function cook(): void {
    if (weight === 0) {
        rl.question("Le poids du riz: ", (answer) => {
            weight = parseInt(answer);
            if (weight > maxWeight) {
                console.log("Y a beaucoup trop de riz");
                rl.close();
            } else {
                rl.question("Pour ajouter de l'eau dans le rice cooker, cliquer sur enter", () => {
                    rl.question("Si vous voulez faire cuire tout de suite, cliquer sur le nombre 0 sinon veuillez indiquer dans combien de minute vous voulez commencer la cuisson: ", (startInput) => {
                        start = parseInt(startInput);
                        if (start === 0) {
                            powerOne = true;
                            console.log("Cuisson");
                        } else {
                            console.log("La cuisson commencera dans " + start + " minutes");
                        }
                        keepWarm = true;
                        console.log("Maintien au chaud ");
                        //rl.close();
                        processOption();
                    });
                });
            }
        });
        rl.close;
    } else {
        console.log("Vous ne pouvez pas en ajouter");
        rl.close();
    }
}

function prediction(): void {
    if (weight <= 500 && weight > 300) {
        console.log("Votre cuisson sera finie dans " + (45 + start) + "min");
    } else if (weight <= 300 && weight > 100) {
        console.log("Votre cuisson sera finie dans 35 min");
    } else if (weight <= 100 && weight>0) {
        console.log("Votre cuisson sera finie dans 20 min");
    } else {
        console.log("Y a même pas de riz dans le rice cooker");
    }
    prediction();
}

function processOption(): void {
    rl.question("Entrez votre choix : ", (answer) => {
        let option: number = parseInt(answer);
            
        switch (option) {
            case 1:
                cook();
                //processOption();
                break;
            case 2:
                prediction();
                //processOption();
                break;
            case 3:
                rl.close();
                break;
            default:
                console.log("Option invalide.");
                processOption();
        }   
    });
    
}

processOption();

File: cooker.py
powerOne = False
keepWarm = False  # quand cook est fini avec succes
# en g
maxWeight = 500
weight = 0
start = 0


def cook():
    global weight, powerOne, keepWarm, start
    if weight == 0:
        weight = int(input("Le poids du riz : "))
        if weight > maxWeight:
            print("Y a beaucoup trop de riz")
        else:
            input("Pour ajouter de l'eau dans le rice cooker, cliquer sur enter")
            start = input(
                "Si vous voulez faire cuire tout de suite, cliquer sur le nombre 0 sinon veuillez indiquer dans combien de minute vous voulez commencer la cuisson : ")
            if int(start) == 0:
                powerOne = True
                print("Cuisson")
            else:
                print("La cuisson commencera dans " + start + " minutes")
            keepWarm = True
            print("Maintien au chaud")
    else:
        print("Vous pouvez pas en ajouter")


def prediction():
    if 500 >= weight > 300:
        print("Votre cuisson sera finie dans " + str(45 + int(start)) + " min")
    elif 300 >= weight > 100:
        print("Votre cuisson sera finie dans "+str(30 + int(start))+" min")
    elif 0<weight <= 100:
        print("Votre cuisson sera finie dans "+str(20 + int(start))+"min")
    else:
        print("Y a même pas de riz dans le rice cooker")


option = int(input("Entrez votre choix : "))
while option != 3:
    if option == 1:
        cook()
    elif option == 2:
        prediction()
    elif option == 3:
        break
    else:
        print("Option invalide.")
    option = int(input("Entrer un autre choix : "))
